### PR TITLE
Refactor clients

### PR DESCRIPTION
**Pull request by tiagonapoli**
#### What is the purpose of this pull request?
Clients logic were scattered all around toolbelt - now they're centralized and compartimentalized 

#### How should this be manually tested?
```
git checkout chore/refactor-clients

yarn watch
```
Play around with the commands

I will ask dev-tools team to use the beta released for this version.

#### Types of changes
- [x] Refactor (non-breaking change that only makes the code better)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [x] Update `CHANGELOG.md`

### CODE REVIEW

File: src/lib/clients/IOClients/IOClientFactory.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      logger: IOClientFactory.createDummyLogger(),
+    }
+  }
+
+  public static createClient<T extends IOClient>(
+    ClientClass: typeof IOClient,
+    customContext: Partial<IOContext> = {},
+    customOptions: Partial<InstanceOptions> = {}
+  ): T {
+    const clusterHeader = env.cluster() ? { 'x-vtex-upstream-target': env.cluster() } : null
```
Comment (Jeymisson) [inline src/lib/clients/IOClients/IOClientFactory.ts:67]: Since we're refactoring the clients. I would start caching header keys on a `constants` file.

Comment (tiagonapoli) [reply src/lib/clients/IOClients/IOClientFactory.ts:67]: Perfect! Will do, and that is much better

Comment (tiagonapoli) [reply src/lib/clients/IOClients/IOClientFactory.ts]: Done :snowboarder:

File: src/lib/constants/Headers.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export enum Headers {
+  VTEX_ORIGINAL_CREDENTIAL = 'x-vtex-original-credential',
+  VTEX_RUNTIME_API = 'x-vtex-runtime-api',
+  VTEX_STICKY_HOST = 'x-vtex-sticky-host',
+  VTEX_UPSTREAM_TARGET = 'x-vtex-upstream-target',
```
Comment (tiagonapoli) [inline src/lib/constants/Headers.ts:5]: Should I add a little doc on each? Like:
https://github.com/vtex/toolbelt/blob/ec41288edb94a5e8b3b527c151228985c73c18b6/src/lib/constants/EnvVariables.ts#L1-L10

Comment (Jeymisson) [reply src/lib/constants/Headers.ts:5]: Sounds good! I'd like that

**Pull request merged**